gremlin-go/driver/client.go

```diff
@@ -71,17 +75,14 @@ func (client *Client) Close() error {
 // Submit submits a Gremlin script to the server and returns a ResultSet
 func (client *Client) Submit(traversalString string) (ResultSet, error) {
 	// TODO AN-982: Obtain connection from pool of connections held by the client.
+	client.logHandler.logf(Debug, submitStartedString, traversalString)
 	request := makeStringRequest(traversalString)
-	if client.connection == nil {
-		client.connection = &connection{
-			host:            client.host,
-			port:            client.port,
-			transporterType: client.transporterType,
-			logHandler:      client.logHandler,
-			transporter:     nil,
-			protocol:        nil,
-		}
-	}
+	return client.connection.write(&request)
+}
 
+// SubmitBytecode submits bytecode to the server to execute and returns a ResultSet
```

Comment:
nitpick: ideally doc comments (according to https://go.dev/blog/godoc) should be sentences ended with proper punctuation.